src/actions.rs

```diff
@@ -100,6 +101,22 @@ impl ActionHandler {
                                 continue;
                             }
                             let span = message.spans[0].rls_span().zero_indexed();
+
+                            // build a more sophisticated error message
+                            let mut message_text = message.message.clone();
+                            for sp in &message.spans {
+                                if sp.is_primary && sp.label.is_some() {
+                                    message_text.push_str("\n⇒ ");
```

Comment:
I'd prefer to keep strings to ascii (maybe use `->` or `=>`) - playing safe since we want to work with any and all editors and can't guarantee they'll handle unicode properly